erpnext/accounts/doctype/journal_entry_template/journal_entry_template.js

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on("Journal Entry Template", {
+	// refresh: function(frm) {
+
+	// }
+	setup: function(frm) {
+		frm.set_query("account" ,"accounts", () => {
+				return {
+					filters: {
+						"company": frm.doc.company==undefined ? null: frm.doc.company,
```

Comment:
```suggestion
						"company": frm.doc.company,
```
Make link field for company mandatory, So user will have to enter company and also default company will be fetched from global defaults  so undefined check is not required